File: codegen/codegen-elixir/src/test/resources/ir/prefix_headers_from_list.expected.ex
defp prefix_headers_from_list(
  headers,
  prefix
) do
  headers
  |> Enum.filter(fn {name, _} -> String.starts_with?(name, prefix) end)
  |> Map.new(fn {name, val} -> {String.slice(name, byte_size(prefix)..-1//1), val} end)
  |> case do
    map when map == %{} -> nil
    map -> map
  end
end

File: codegen/codegen-erlang/src/test/resources/ir/prefix_headers_from_list.expected.erl
prefix_headers_from_list(Headers, Prefix) ->
    Map = maps:from_list([
        {binary:part(Name, byte_size(Prefix)), Val}
     || {Name, Val} <- Headers,
        byte_size(Name) > byte_size(Prefix),
        binary:part(Name, 0, byte_size(Prefix)) =:= Prefix
    ]),
    case maps:size(Map) of
        0 -> undefined;
        _ -> Map
    end.
